synapse/daemon.py

```diff
@@ -181,6 +180,11 @@ async def t2call(link, meth, args, kwargs):
             else:
                 logger.exception('error during task %s', meth.__name__)
 
+            if isinstance(valu, types.AsyncGeneratorType):
```

Comment:
Should this be done in a finally block?

Reply:
i guess it depends on what happens if you call close on a closed generator

Reply:
Just seems like we would want to be careful about when we are closing, in the event that closing a given generator throws (scary in and of itself); but closing them after we've sent the next lines feels safer for some reason. Multiple close calls should be safe.

Reply:
There's no point is double closing a generator.

If a generator throws after being closed, that's a violation of the generator spec (and would normally be a RuntimeError).

I didn't want to close the link first as the closing generator might do something with it.